src/App.tsx

```diff
@@ -1,14 +1,86 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+
+const prepareTodos = (todos: Todo[], filter = FilterBy.all, query: string) => {
+  let todosToPrepare = [...todos];
+
+  if (filter) {
+    switch (filter) {
+      case FilterBy.active:
+        todosToPrepare = todosToPrepare.filter(todo => !todo.completed);
+        break;
+      case FilterBy.completed:
+        todosToPrepare = todosToPrepare.filter(todo => todo.completed);
+        break;
+      case FilterBy.all:
+        todosToPrepare = [...todos];
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (query) {
+    const lowerQuery = query.toLowerCase();
+
+    todosToPrepare = todosToPrepare.filter(todo => todo.title
+      .toLowerCase()
+      .includes(lowerQuery));
+  }
+
+  return todosToPrepare;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [areTodosLoading, setAreTodosLoading] = useState(true);
```

Comment:
Just `isLoading` is enough, it's common name of loading state, and you can use the same name for state in other components